insights/parsers/nftables.py

```diff
@@ -0,0 +1,154 @@
+                                    "family": "ip", "table": "example_table1", "chain": "example_chain1", "handle": 3, "expr": [{"vmap": {"key": {"payload": {"protocol": "ip", "field": "saddr"}}, "data": "@example_map"}}]
+                                },
+                                {
+                                    "family": "ip", "table": "example_table", "chain": "example_chain", "handle": 7, "expr": [{"match": {"op": "==", "left": {"payload": {"protocol": "tcp", "field": "dport"}}, "right": 22}}, {"counter": {"packets": 29, "bytes": 1976}}, {"accept": None}]
+                                }
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+
+    Attributes:
+        main_data (dict): A dict contains the address_family/table/chain/rule data.
+
+    Raises:
+        SkipComponent: No available nftables.
+
+    Examples:
+        >>> type(nft_obj)
+        <class 'insights.parsers.nftables.NftListRuleSet'>
+        >>> [str(key) for key in nft_obj.main_data.keys()]  # to be compatible with python2.6/2.7, transform unicode to string
+        ['ip']
+        >>> [str(key) for key in nft_obj.main_data['ip'].keys()] # to be compatible with python2.6/2.7, transform unicode to string
+        ['example_table1']
+        >>> 'example_chain1' in nft_obj.chains('ip', 'example_table1')
+        True
+        >>> len(nft_obj.rules('ip', 'example_table1', 'example_chain1'))
+        2
+
+    """
+
+    def parse_content(self, content):
+        """
+        There are a lot of other types in nftables besides "chain/rule", like "map", "set", "quota".
+        But they are not extracted now since there are no rules dependent on them. If needed,
+        we can extract them later.
+        """
+        super(NftListRuleSet, self).parse_content(content)
+        self.main_data = defaultdict(dict)
+        for item in self.data['nftables']:
+            for key, value in item.items():
+                address_name = value.get('family')
+                if key == 'table':
+                    table_name = value['name']
+                    self.main_data[address_name][table_name] = defaultdict(dict)
+                else:
+                    if key in ['chain', 'rule']:
+                        table_name = value['table']
+                        if key == 'chain':
+                            chain_name = value['name']
+                            self.main_data[address_name][table_name]['chains'][chain_name] = {}
+                            self.main_data[address_name][table_name]['chains'][chain_name]['attrs'] = value
+                            self.main_data[address_name][table_name]['chains'][chain_name]['rules'] = []
+                        else:
+                            # rule
+                            chain_name = value['chain']
+                            self.main_data[address_name][table_name]['chains'][chain_name]['rules'].append(value)
```

Comment:
What about simplifying the key check in this if ... else block to the same level, say :  
```
                if key == 'table':
                   ...
                elif key == 'chain':
                    ...
                elif key == 'rule':
                    ...
```